### ISSUE TITLE
Binary .prob file for more efficient processing

### ISSUE THREAD

**Issue by ahonkela** [BitSeq/BitSeq#14]
We should change to binary .prob format for more efficient processing. (Big text files of floats are inefficient!)

**Comment by ahonkela** [comment]
After some thought, I am starting to think that a reasonably engineered custom binary format plus converters to turn it to text is probably the best solution. If we go with the reader/writer API, the converter should be fairly trivial to implement anyway. I already refactored the writer into a separate class last night. (The code is in branch probformats.) The interface may need tuning to make it easier to use a binary format with parallel access.

I wrote a draft of binary .prob (.bprob?) file format specification:
https://github.com/BitSeq/BitSeq/blob/probformats/docs/bprob_file_format.md

This may be too ambitious, but at least I believe it should be fairly convenient for parallel access and would also clarify things when we could get rid of the separate .tr files.

Antti
